rpm/dtc.spec

```diff
@@ -1,45 +1,82 @@
+%define     __dtc_version     1.4.6
+
 Name:       dtc
-Summary:    Linux Kernel device tree compiler
-Version:    0.0.0
-Provides:   dtc
+Version:    %{__dtc_version}
 Release:    1
+Summary:    Device Tree Compiler
+License:    GPLv2+
 Group:      Development/Tools
-URL:        https://git.kernel.org/pub/scm/utils/dtc/dtc.git
+URL:        https://devicetree.org/
+
 Source0:    %{name}-%{version}.tar.bz2
-License:    GPLv2
 
-BuildRequires: flex
-BuildRequires: bison
+BuildRequires:     gcc
+BuildRequires:     make
+BuildRequires:     flex
+BuildRequires:     bison
 
 %description
-Linux Kernel device tree compiler
+Devicetree is a data structure for describing hardware. Rather than hard coding
+every detail of a device into an operating system, many aspects of the hardware
+can be described in a data structure that is passed to the operating system at
+boot time. The devicetree is used by OpenFirmware, OpenPOWER Abstraction Layer
+(OPAL), Power Architecture Platform Requirements (PAPR) and in the standalone
+Flattened Device Tree (FDT) form.
+
+%package -n libfdt
+Summary: Device tree library
+Requires(post): /sbin/ldconfig
+Requires(postun): /sbin/ldconfig
+
+%description -n libfdt
+libfdt is a library to process Open Firmware style device trees on various
+architectures.
+
+%package -n libfdt-devel
+Summary: Development headers for device tree library
+Requires: libfdt = %{version}-%{release}
+
+%description -n libfdt-devel
+This package provides development files for libfdt
+
+%package -n libfdt-static
+Summary: Static version of device tree library
+Requires: libfdt-devel = %{version}-%{release}
+
+%description -n libfdt-static
+This package provides the static library of libfdt
 
 %prep
-%setup -q
+%autosetup -p1
 
 %build
 cd dtc
-# We only need the compiler, no python necessary so skip it.
-make NO_PYTHON=1 HOME=$RPM_BUILD_ROOT/usr
+NO_PYTHON=1 make %{?_smp_mflags} V=1 CC="gcc $RPM_OPT_FLAGS $RPM_LD_FLAGS"
 
 %install
 cd dtc
-make NO_PYTHON=1 HOME=$RPM_BUILD_ROOT/usr install
+NO_PYTHON=1 make install DESTDIR=$RPM_BUILD_ROOT SETUP_PREFIX=$RPM_BUILD_ROOT/usr PREFIX=/usr LIBDIR=%{_libdir}
+mkdir -p ${RPM_BUILD_ROOT}/%{_defaultlicensedir}/%{name}
+mkdir -p ${RPM_BUILD_ROOT}/%{_defaultdocdir}/%{name}
+install GPL ${RPM_BUILD_ROOT}/%{_defaultlicensedir}/%{name}/
+install Documentation/manual.txt ${RPM_BUILD_ROOT}/%{_defaultdocdir}/%{name}/
+
+%post -n libfdt -p /sbin/ldconfig
+%postun -n libfdt -p /sbin/ldconfig
 
 %files
-%defattr(-,root,root,-)
-%{_bindir}/convert-dtsv0
-%{_bindir}/dtc
-%{_bindir}/dtdiff
-%{_bindir}/fdtdump
-%{_bindir}/fdtget
-%{_bindir}/fdtoverlay
-%{_bindir}/fdtput
-%{_includedir}/fdt.h
-%{_includedir}/libfdt.h
-%{_includedir}/libfdt_env.h
-%{_libdir}/libfdt-1.4.6.so
+%attr(0644, -, -) %{_defaultlicensedir}/%{name}/GPL
+%attr(0644, -, -) %{_defaultdocdir}/%{name}/manual.txt
+%{_bindir}/*
+
+%files -n libfdt
+%attr(0644, -, -) %{_defaultlicensedir}/%{name}/GPL
```

Comment:
As well this, however if the main package depends on this package it is enough to have this license only.